File: 0189-rotate-array/solution.java
class Solution {
    public void rotate(int[] nums, int k) {
        int[] arr= new int[nums.length*2];
        k=k%nums.length;
        k=nums.length-k;
        for(int i=0;i<nums.length;i++)
        {
            arr[i]=nums[i];
            arr[i+nums.length]=nums[i];
        }
        for(int i=k;i<k+nums.length;i++)
        {
            nums[i-k]=arr[i];
        }
    }
}

File: 0189-rotate-array/solution.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n=nums.size();
        k%=n;
        reverse(0,n-1,nums);
        reverse(0,k-1,nums);
        reverse(k,n-1,nums);
    }
    void reverse(int s ,int e ,vector<int>& nums ){
        while(s<e){
            int temp=nums[s];
            nums[s]=nums[e];
            nums[e]=temp;
            s++;
            e--;
        }
    }
};
